ib_manifest_util/util.py

```diff
@@ -136,3 +142,83 @@ def download_files(urls: str | list) -> list:
         logger.info(f"Downloaded file {filename} from {address}.")
 
     return filenames
+
+
+def verify_local_channel_environments(
+    local_channel_env: str | Path,
+    offline_mode: bool = True,
+    conda_binary_loc: str | Path | None = None,
+):
+    """
+    Verify that the conda environment (`local_channel_env`) can be successfully created.
+
+    Args:
+        local_channel_env: path to local environment yaml
+        offline_mode: whether to run `conda env create` with the `--offline` flag, default is true
+        conda_binary_loc: path to a `conda` binary, default will try to locate the `conda` binary by using the return value of `$ which conda`
+
+    NOTE: It is assumed that the `conda-vendor vendor` command has already been run and that the `local_channel_env`
+    has a single channel set to `path/to/local_channel/folder`.
+    """
+
+    logger.info(
+        "Verifying that the local channel environment can be successfully created"
+    )
+    if isinstance(local_channel_env, str):
+        local_channel_env = Path(local_channel_env)
+
+    if not local_channel_env.exists():
+        raise ValueError(
+            f"The environment file submitted appears not to exist at location: {local_channel_env.resolve()}"
+        )
+
+    env_yaml = load_yaml(local_channel_env)
+    channels = env_yaml["channels"]
+    name = env_yaml["name"]
+
+    if len(channels) > 1 or not Path(channels[0]).exists():
+        raise ValueError(
+            f"The `channels` key for {local_channel_env} is misformatted: {channels}. It should only contain one item, the file path to the local channel folder"
+        )
+
+    if conda_binary_loc:
+        if isinstance(conda_binary_loc, str):
+            conda_binary_loc = Path(conda_binary_loc)
+        if not conda_binary_loc.exists():
+            raise ValueError(
+                f"The `conda` binary provided appears not to exist at location: {conda_binary_loc.resolve()}"
+            )
+        conda_binary = str(conda_binary_loc)
+    else:
+        try:
+            # determine location of conda binary
+            conda_binary = run_subprocess("which conda", return_as_str=True).strip("\n")
+        except Exception as e:
+            logger.warning(
+                "Having trouble determine the location of the `conda` binary, falling back to simply using `conda`"
```

Comment:
if `which conda` fails, won't every command that calls `conda` fail?

Reply:
if we get down this path, then `conda_binary` is undefined. I think you need a `conda_binary = "conda"` here.... but I also think this branch should be a true failure.

Reply:
Yeah I think you're ultimately right. I had originally added to in-case the `run_subcommand` function was acting up but if that's the case, it will fail when it tries to create the conda env.